### PR TITLE
#24-created-new-random-event-class

### PR DESCRIPTION
**Pull request by LaurenEBaker**
this was just to create the new class that extends events, I'm just slowly pushing it little by little instead of all at once.

### CODE REVIEW

**Comment by JMarlin** [comment]
I respect the hustle and your thought process is pretty right on. I would normally say don't make a whole PR, just let me know when you want me to take a look and I can just look at the commits on your feature branch. 

But since you already have the PR made, you can just keep pushing changes to that branch and it should update the PR as well. So I guess we can use it as a communication tool.

**Comment by LaurenEBaker** [comment]
I figured this way you could see what I have accomplished so far, along with some issues I've ran into. Mainly on line 40 of the main method. I changed the class type for RandomSpawner from abstract to not have that, but main is throwing the error that the constructor is undefined. I'm also having trouble understanding what you mean by:
 "For each case in the current switch, create a corresponding stub immediately triggered event. (the triggerAction() of the event can just do the println that's currently in each switch case)"
Would it be possible to reword what you mean by "stub immediately triggered event"?

**Comment by JMarlin** [comment]
Just adding a note that @LaurenEBaker and I met on Friday and worked through some of the above questions. Should have good direction now

File: src/main/java/com/corporate/hellscape/events/Nothing.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+	private boolean _triggerEnabled;
+    public Nothing(boolean triggerEnabled) {
+        _triggerEnabled = _triggerEnabled; }
+    public boolean isTriggered(Hellscape hellscape) {
+        return _triggerEnabled;}
+    public void triggerAction(Hellscape hellscape) {
+        System.out.println("Nothing eventful happened");
+		Character.Health= Character.getHealth;
+
+        hellscape.registerEvent(new Nothing(false));
```
Comment (JMarlin) [inline src/main/java/com/corporate/hellscape/events/Nothing.java:14]: No need to register a follow-up event, that should just be handled by the `RandomEvent`

File: src/main/java/com/corporate/hellscape/events/Spider.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package com.corporate.hellscape.events;
+import com.corporate.hellscape.Hellscape;
+
+public class Spider extends Event {
+	private boolean _triggerEnabled;
+
+    public Spider(boolean triggerEnabled) {
+        _triggerEnabled = _triggerEnabled; }
```
Comment (JMarlin) [inline src/main/java/com/corporate/hellscape/events/Spider.java:8]: Should probably be `_triggerEnabled = triggerEnabled`

File: src/main/java/com/corporate/hellscape/events/Spider.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    public boolean isTriggered(Hellscape hellscape) {
+        return _triggerEnabled;}
+    public void triggerAction(Hellscape hellscape) {
+        System.out.println("you were bitten by a posionous spider");
+		Character.Health =- 10;
+		Character.Stress =+ 5;
+		Character.Fun =- 5;
+
+        hellscape.registerEvent(new Spider(false));
```
Comment (JMarlin) [inline src/main/java/com/corporate/hellscape/events/Spider.java:17]: No need to register a follow-up event

**Review by JMarlin — CHANGES_REQUESTED** [review]

File: src/main/java/com/corporate/hellscape/events/RandomEvent.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+		SetDelay(ran.nextInt(), hellscape);
+	}
+	
+
+	@Override
+	protected void triggerAction(Hellscape hellscape) {
+		Random rand = new Random();
+		int num1 = rand.nextInt(3); switch(num1%3) {
+	case 0: 
+		hellscape.registerEvent(Event.Gift);
```
Comment (JMarlin) [inline src/main/java/com/corporate/hellscape/events/RandomEvent.java:19]: I'm not sure what this line is supposed to do

File: src/main/java/com/corporate/hellscape/events/RandomEvent.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+		hellscape.registerEvent(new Gift(false));
+		break;
+	case 1: 
+		hellscape.registerEvent(Event.Spider);
```
Comment (JMarlin) [inline src/main/java/com/corporate/hellscape/events/RandomEvent.java:23]: I'm not sure what this line is supposed to do

File: src/main/java/com/corporate/hellscape/events/RandomEvent.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+		hellscape.registerEvent(new Spider(false));
+		break;
+	case 2: 
+		hellscape.registerEvent(Event.Nothing);
```
Comment (JMarlin) [inline src/main/java/com/corporate/hellscape/events/RandomEvent.java:27]: I'm not sure what this line is supposed to do

**Review by JMarlin — CHANGES_REQUESTED** [review]

**Review by JMarlin — APPROVED** [review]

**Pull request merged**